src/drt/src/pa/FlexPA_prep.cpp

```diff
@@ -499,148 +499,87 @@ void FlexPA::genAPsFromRect(std::vector<std::unique_ptr<frAccessPoint>>& aps,
-                     true);
-        genAPOnTrack(x_coords,
-                     layer1_track_coords,
-                     gtl::xl(rect) - min_width_layer1,
-                     gtl::xl(rect),
-                     true);
-      }
-      if (upper_type >= frAccessPointEnum::NearbyGrid) {
-        genAPOnTrack(y_coords,
-                     layer2_track_coords,
-                     gtl::yh(rect),
-                     gtl::yh(rect) + min_width_layer2,
-                     true);
-        genAPOnTrack(y_coords,
-                     layer2_track_coords,
-                     gtl::yl(rect) - min_width_layer2,
-                     gtl::yl(rect),
-                     true);
-      }
+                   layer2_rect_min - min_width_layer2,
+                   layer2_rect_min,
+                   true);
     }
   } else {
-    if (is_layer1_horz) {
-      lower_type = frAccessPointEnum::OnGrid;
-      genAPOnTrack(x_coords, layer2_track_coords, gtl::xl(rect), gtl::xh(rect));
-      if (upper_type >= frAccessPointEnum::Center) {
-        genAPCentered(x_coords, layer_num, gtl::xl(rect), gtl::xh(rect));
-      }
-      if (upper_type >= frAccessPointEnum::EncOpt) {
-        genAPEnclosedBoundary(x_coords, rect, layer_num, !is_layer1_horz);
-      }
-      if (upper_type >= frAccessPointEnum::NearbyGrid) {
-        genAPOnTrack(x_coords,
-                     layer2_track_coords,
-                     gtl::xh(rect),
-                     gtl::xh(rect) + min_width_layer2,
-                     true);
-        genAPOnTrack(x_coords,
-                     layer2_track_coords,
-                     gtl::xl(rect) - min_width_layer2,
-                     gtl::xl(rect),
-                     true);
-      }
-      genAPCentered(y_coords, layer_num, gtl::yl(rect), gtl::yh(rect));
-      for (auto& [y_coord, cost] : y_coords) {
-        y_coords[y_coord] = frAccessPointEnum::OnGrid;
-      }
-    } else {
-      genAPOnTrack(y_coords, layer2_track_coords, gtl::yl(rect), gtl::yh(rect));
-      if (upper_type >= frAccessPointEnum::Center) {
-        genAPCentered(y_coords, layer_num, gtl::yl(rect), gtl::yh(rect));
-      }
-      if (upper_type >= frAccessPointEnum::EncOpt) {
-        genAPEnclosedBoundary(y_coords, rect, layer_num, !is_layer1_horz);
-      }
-      if (upper_type >= frAccessPointEnum::NearbyGrid) {
-        genAPOnTrack(y_coords,
-                     layer2_track_coords,
-                     gtl::yh(rect),
-                     gtl::yh(rect) + min_width_layer2,
-                     true);
-        genAPOnTrack(y_coords,
-                     layer2_track_coords,
-                     gtl::yl(rect) - min_width_layer2,
-                     gtl::yl(rect),
-                     true);
-      }
-      genAPCentered(x_coords, layer_num, gtl::xl(rect), gtl::xh(rect));
-      for (auto& [x_coord, cost] : x_coords) {
-        x_coords[x_coord] = frAccessPointEnum::OnGrid;
-      }
+    lower_type = frAccessPointEnum::OnGrid;
```

Comment:
There is an odd asymmetry to this line.  It only appears in one of the four branches.  I think this PR moves it into two of the branches.  I'm not sure it belongs in any.